__tests__/workers/notifications/collectionUpdated.ts

```diff
@@ -0,0 +1,165 @@
+      yggdrasilId: '5a829977-189a-4ac9-85cc-9e822cc7c737',
+    },
+    {
+      id: 'cp3',
+      shortId: 'cp3',
+      url: 'http://cp3.com',
+      score: 0,
+      metadataChangedAt: new Date('01-05-2020 12:00:00'),
+      sourceId: 'c',
+      visible: true,
+      createdAt: new Date('01-05-2020 12:00:00'),
+      origin: PostOrigin.Crawler,
+      yggdrasilId: '5a829977-189a-4ac9-85cc-9e822cc7c737',
+    },
+    {
+      id: 'cp4',
+      shortId: 'cp4',
+      url: 'http://cp4.com',
+      score: 0,
+      metadataChangedAt: new Date('01-05-2020 12:00:00'),
+      sourceId: 'community',
+      visible: true,
+      createdAt: new Date('01-05-2020 12:00:00'),
+      origin: PostOrigin.Crawler,
+      yggdrasilId: '5a829977-189a-4ac9-85cc-9e822cc7c737',
+    },
+  ]);
+  await saveFixtures(con, CollectionPost, [
+    {
+      id: 'c1',
+      shortId: 'c1',
+      title: 'My collection',
+      score: 0,
+      metadataChangedAt: new Date('01-05-2020 12:00:00'),
+      sourceId: 'a',
+      visible: true,
+      createdAt: new Date('01-05-2020 12:00:00'),
+      yggdrasilId: '7ec0bccb-e41f-4c77-a3b4-fe19d20b3874',
+    },
+  ]);
+  await saveFixtures(con, PostRelation, [
+    {
+      postId: 'c1',
+      relatedPostId: 'cp1',
+      type: PostRelationType.Collection,
+    },
+    {
+      postId: 'c1',
+      relatedPostId: 'cp2',
+      type: PostRelationType.Collection,
+    },
+    {
+      postId: 'c1',
+      relatedPostId: 'cp3',
+      type: PostRelationType.Collection,
+    },
+    {
+      postId: 'c1',
+      relatedPostId: 'cp4',
+      type: PostRelationType.Collection,
+    },
+  ]);
+
+  await saveFixtures(con, NotificationPreferencePost, [
+    {
+      userId: '1',
+      referenceId: 'c1',
+      notificationType: NotificationType.CollectionUpdated,
+      status: NotificationPreferenceStatus.Subscribed,
+    },
+    {
+      userId: '2',
+      referenceId: 'c1',
+      notificationType: NotificationType.CollectionUpdated,
+      status: NotificationPreferenceStatus.Subscribed,
+    },
+    {
+      userId: '3',
+      referenceId: 'c1',
+      notificationType: NotificationType.CollectionUpdated,
+      status: NotificationPreferenceStatus.Subscribed,
+    },
+  ]);
+});
+
+it('should notifiy when a collection is updated', async () => {
+  const actual = await invokeNotificationWorker(worker, {
+    post: {
+      id: 'c1',
+      title: 'My collection',
+      content_type: PostType.Collection,
+    },
+  });
+
+  expect(actual.length).toEqual(3);
+
+  actual.forEach((bundle) => {
+    const ctx = bundle.ctx as NotificationCollectionContext;
+    expect(bundle.type).toEqual('collection_updated');
+    expect(ctx.post.id).toEqual('c1');
+    expect(ctx.distinctSources.length).toEqual(3);
+    expect(ctx.total).toEqual('4');
+  });
+  expect(actual.map((bundle) => bundle.ctx.userId)).toEqual(['1', '2', '3']);
+
+  expect(
+    (actual[0].ctx as NotificationCollectionContext).distinctSources[0].name,
+  ).toEqual('A');
```

Comment:
Maybe also verify the full content of notification like title, text, basically big toMatchObject check against notification object.

Reply:
Added in 07c29d82ee3b5bb7b016687e31324eee6946b0ac